time/timepersist/src/time_persist_nvreg.c

```diff
@@ -0,0 +1,75 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *  http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+#include <os/mynewt.h>
+
+#if MYNEWT_VAL(TIMEPERSIST_NVREG_INDEX) >= 0
+#if MYNEWT_VAL(TIMEPERSIST_SYS_CONFIG)
+#error "Can't have both methods of persisting"
+#endif
```

Comment:
Is this required if you already added a restriction on the syscfg?

Reply:
Strictly speaking no. I've had mixed success previously when writing restrictions.
But I guess I can remove this check, as the build would fail eventually due to duplicate symbols, if both ended up defined at the same time :)